src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateDtoValidator.java

```diff
@@ -194,6 +195,25 @@ protected void validateAttributeAndNameCodes(final BindingResult errors, final S
 		}
 	}
 
+	protected void validateCannotAssignSelfAsProgenitor(final BindingResult errors, final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
```

Comment:
Hi @cheryleigh 
My concern is we are going 2 times to query by GUID (see this.validateGermplasmIdAndGermplasmUUID) 
I think we can split `validateGermplasmIdAndGermplasmUUID` and return the list og germplasm as optional and use that list to run a 2nd validation 
I dont see any risk on doing this improvement now. If you find it complicated, let me know and we can create a follow up task

Reply:
Hi @clarysabel ,

I updated the implementation to retrieve the list first and pass it as parameter to the 2 methods. Kindly check if this is alright. Thank you!